mirror-maker-2/active-passive-ssl/deployment.yaml

```diff
@@ -0,0 +1,109 @@
+              gomplate -f /config-template/connect-mirror-maker-template.properties -o ./config/connect-mirror-maker.properties
+
+          env:
+            - name: SOURCE_NAME
+              value: source
+            - name: SOURCE_BOOTSTRAP_SERVERS
+              value: source:9092
+            - name: TARGET_NAME
+              value: target
+            - name: TARGET_BOOTSTRAP_SERVERS
+              value: target:9092
+            - name: TOPICS
+              value: ".*"
+            - name: GROUPS
+              value: ".*"
+            # if set to true, the source name will be added as a prefix to the replicated topics in the target cluster.
+            # Example: with source name set to 'otel', when replicating topic otlp_spans, the topic name in the target will be 'otel.otlp_spans'
+            - name: TARGET_TOPICS_ADD_CLUSTER_PREFIX
+              value: "false"
+            - name: SYNC_TOPIC_CONFIGS
+              value: "true"
+            - name: SYNC_TOPIC_ACLS
+              value: "false"
+            # When source has SSL enabled, mirror maker uses JKS keystore and truststore encrypted by password to connect. Use properties below
+            - name: SOURCE_SSL_ENABLE
+              value: "false"
+            # The location to the JKS truststore
+            - name: SOURCE_TRUSTSTORE
+            - name: SOURCE_TRUSTSTORE_PASSWORD
+            # The location to the JKS keystore
+            - name: SOURCE_KEYSTORE
+            - name: SOURCE_KEYSTORE_PASSWORD
+
+            # When target has SSL enabled, mirror maker uses JKS keystore and truststore encrypted by password to connect. Use properties below
+            - name: TARGET_SSL_ENABLE
+              value: "false"
+            # The location to the JKS truststore
+            - name: TARGET_TRUSTSTORE
+            - name: TARGET_TRUSTSTORE_PASSWORD
+            # The location to the JKS keystore
+            - name: TARGET_KEYSTORE
+            - name: TARGET_KEYSTORE_PASSWORD
+          volumeMounts:
+            - name: mm2-cfg
+              mountPath: /config
+            - name: mm2-cfg-template
+              mountPath: /config-template
+      containers:
+        - name: *app
+          imagePullPolicy: IfNotPresent
+          image: bitnami/kafka:3.4
+          env:
+            - name: KAFKA_HEAP_OPTS
+              value: -Xmx4G -Xms1G
+          command:
+            #  the config is mounted from file generated from a template in the init container
+            - sh
+            - -ecx
+            - /opt/bitnami/kafka/bin/connect-mirror-maker.sh /config/connect-mirror-maker.properties
+          lifecycle:
+            # stop the java process with a prestop hook, otherwise the TERM signal doesn't get to the process, since it is spawned by the connect-mirror-maker.sh
+            preStop:
+              exec:
+                command: ["/bin/sh", "-c", "kill $(ps -ef | grep java | grep -v grep | awk '{print $2}')"]
```

Comment:
:thinking: it looks like the [`connect-mirror-maker.sh `](https://github.com/apache/kafka/blob/de409b389d26f7681fba8583db2b96584258aa48/bin/connect-mirror-maker.sh#L45) script will `exec` so I would expect any signals sent to that process to be passed to the Java process. Is the signal behaviour we're seeing maybe due to the fact the container's comment above is `sh -c /script` (vs just `/script`)? In which case process 1 is just `sh` and won't be passing any signals down to the child process

Reply:
Changed it to launch the script and it is working now.